submissions/kasionio/oop-exercise/index.js

```diff
@@ -0,0 +1,76 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/kasionio/a-tiny-JS-world/tree/gh-pages
+   Web app: https://kasionio.github.io/a-tiny-JS-world
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Inhabitant {
+  constructor(species, name, gender, legs, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.saying = saying;
+  }
+}
+
+class Human extends Inhabitant {
+  constructor(name, gender, saying, hands) {
+      super('human', name, gender, 2, saying);
+      this.hands = 2;
+  }
+}
+
+class Man extends Human {
+  constructor(name, saying) {
+      super(name, 'male', saying);
+  }
+}
+
+class Woman extends Human {
+  constructor(name, saying) {
+      super(name, 'female', saying);
+  }
+}
+
+class Animal extends Inhabitant {
+  constructor(species, name, gender, saying) {
+      super(species, name, gender, 4, saying);
+  }
+}
+
+class Dog extends Animal {
+  constructor(name, gender) {
+      super('dog', name, gender, 'woof-woof');
+  }
+}
+
+class Cat extends Animal {
+  constructor(name, gender) {
+      super('cat', name, gender, 'meow');
+  }
+}
+
+class CatWoman extends Inhabitant {
+  constructor(name) {
+      super('catWoman', name, 'female', 2, cat.saying);
```

Comment:
What if `cat` doesn't exist yet?
Besides we want that a cat-woman borrows saying not from a prticular cat but rather from all cats.
Take look into static class properties/methods (and define one in `Cat` class). These are available at all times, exists a particular class instance or not.

Reply:
Thanks for the explanation! It's difficult for me when I go from theory to practice.